neuro_config_client/factories.py

```diff
@@ -814,3 +828,92 @@ def create_ingress(cls, ingress: IngressConfig) -> dict[str, Any]:
         if ingress.cors_origins:
             result["cors_origins"] = ingress.cors_origins
         return result
+
+    @classmethod
+    def create_node_pool(
+        cls, node_pool: NodePool, cloud_provider_type: CloudProviderType
+    ) -> dict[str, Any]:
+        if cloud_provider_type == CloudProviderType.ON_PREM:
+            return cls._create_onprem_node_pool(node_pool)
+        elif cloud_provider_type.value.startswith("vcd_"):
```

Comment:
I was thinking to do so, but now have doubts about it, since the payload differs and is validated differently. For instance:
- idle_size, is_preemptible, zones are allowed in aws/gcp/azure but banned in vcd
- price and currency are banned in aws/gcp/azure, but allowed in vcd
- disk_type is optional for aws/gcp/azure but required for vcd

A single could be overcomplicated with if-else and, basically, we will have this separate method under other `elif cloud_provider_type.value.startswith("vcd_")`
I guess, thereby we validate payload separately on the server side.

wdyt?